src/main/java/utils/SchemaMetaData.java

```diff
@@ -57,17 +110,21 @@ public String getPrimaryKeyColumn(String tableName) {
         try (PreparedStatement ps = connection.prepareStatement(sql)) {
             ps.setObject(1, tableName);
             ResultSet rs = ps.executeQuery();
-            rs.next();
-            columnName = rs.getObject(1, String.class);
+            while(rs.next()) {
+                primaryKeyColumns.add(new ColumnInfo(
+                        rs.getString(1),
+                        rs.getString(2)
+                ));
+            }
             rs.close();
         } catch (SQLException e) {
             throw new RuntimeException("Couldn't get primary key column for table " + tableName + ": " + e.getMessage());
         }
-        return columnName;
+        return primaryKeyColumns;
     }
 
-    public List<ColumnInfo> getForeignKeys(String tableName) {
-        List<ColumnInfo> result = new ArrayList<>();
+    public List<FKColumnInfo> getForeignKeyColumns(String tableName) {
```

Comment:
tu de facto nie otrzymujemy informacji o samej kolumnie z kluczem obcym, a raczej o kluczu obcym - nazwa metody powinna to uwzględniać

Reply:
Nazwę się dopasuje do nowej nazwy zwracanej klasy.